src/main/java/de/tum/in/www1/artemis/web/rest/LearningPathResource.java

```diff
@@ -165,4 +165,35 @@ else if (!authorizationCheckService.isAtLeastInstructorInCourse(course, user) &&
         NgxLearningPathDTO graph = learningPathService.generateNgxGraphRepresentation(learningPath);
         return ResponseEntity.ok(graph);
     }
+
+    /**
+     * GET /courses/:courseId/learning-path-id : Gets the id of the learning path.
+     * If the learning path has not been generated although the course has learning paths enabled, the corresponding learning path will be created.
+     *
+     * @param courseId the id of the course from which the learning path id should be fetched
+     * @return the ResponseEntity with status 200 (OK) and with body the id of the learning path
+     */
+    @GetMapping("/courses/{courseId}/learning-path-id")
```

Comment:
I asked my self the same question regarding the GET/PUT thing. In 99.9% the learning path should already be created by the instructor (when enabling learning paths) or during enrollment. This endpoint will only create it if someone was added to the group outside of the course context (e.g. student was manually added to the group in Jira). Rather than displaying an error/a prompt to contact your instructor, I decided to generate the LP so student's have a better experience.

> And why do we only return an ID?

For now, everything (competencies, lectures, exercises, etc.) is strictly owned by a course. In the future, we are aiming for standardized competencies that are course independent. Furthermore, currently, a learning path is generated for a specific course. This will be extended to Artemis-wide learning paths in the future (one user has exactly one learning path). The design of the endpoints for learning paths already incorporates this vision (as much as possible). Hence, all requests (sent by a user after requesting the LP-id) work with the LP-id to avoid major changes down the line.